scripts/build-controller-image.sh

```diff
@@ -100,16 +105,17 @@ if [[ "$LOCAL_MODULES" = "true" ]]; then
   DOCKERFILE="${ROOT_DIR}"/Dockerfile.local
 fi
 
-if ! docker build \
+if ! docker buildx build \
   --quiet="${QUIET}" \
   -t "${AWS_SERVICE_DOCKER_IMG}" \
   -f "${DOCKERFILE}" \
+  --platform linux/amd64,linux/arm64,linux/ppc64le,linux/s390x,${CONTAINER_OS}/${GOARCH} \
```

Comment:
Ditto what does `--platform` do here? release multiple images for the specified platforms?

Reply:
> release multiple images for the specified platforms?

Yes, it is.
I am using an ARM64 instance in ManagedNodeGroup, and when I tried to start it there, it mocked up.
This is because I don't have an ARM64 image.

Reply:
When you build with this, multiple platforms are described in the manifest file as follows.

└─> docker manifest inspect nginx

```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:9504f3f64a3f16f0eaf9adca3542ff8b2a6880e6abfb13e478cca23f6380080a",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:6967bfc5a04ad3120ada68fe1c89c1bbac92e4b4793e53a5195be314f75ca17b",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v5"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:ab01fca45d6c3ceb8d7b72364eb03a8b47377b5d42cf1433dcf28d365544f1d8",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:85eabf2757cb5b5b84248d7feb019079501dfd8691fc79b8b1d0ff1591a6270b",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:0057a745dbd60ad9c187e89dea876130377a2291df90f28e5fdaaad9c335ae01",
         "platform": {
            "architecture": "386",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:9015ea6b47d4d8d86d4bbd1b0f1e568b56cd703acbd655854716ac6b79c52b1d",
         "platform": {
            "architecture": "mips64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:8e0cd0665c6ad7abd9e421d65f826af933cbc7647b4752f354f260a74b92acf9",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1778,
         "digest": "sha256:968ca27471c39e881787859f77ca757690f2c81bfe87c83384853e2d4f8861a8",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

Reply:
I am also working elsewhere to get the world's container images to work on ARM64.
https://github.com/knative/docs/pull/5273

Reply:
`buildx` will build multiple architectures, but also more importantly builds a `manifest-list` media type, that references all of the architectures available for a given image. When a `manifest-list` is stored in an OCI container registry, and an oci engine (podman, docker, crio, etc) goes to pull the image, the registry returns the manifest-list, then the client(podman) sees if the architecture that is running the engine is present (ie if you are running podman on an arm64 system), and arm64 is listed the client then will request the `blobs` for the arm64 image.

Reply:
Thx 🙏